overlord/devicestate/devicemgr.go

```diff
@@ -2130,21 +2130,21 @@ func (m *DeviceManager) SystemAndGadgetAndEncryptionInfo(wantedSystemLabel strin
 // includes system, gadget information, gadget and kernel snaps info,
 // and gadget and kernel seed snap info.
 func (m *DeviceManager) loadSystemAndEssentialSnaps(wantedSystemLabel string, types []snap.Type) (
-	*System, map[snap.Type]*snap.Info, map[snap.Type]*seed.Snap, error) {
+	*System, seed.Seed, map[snap.Type]*snap.Info, map[snap.Type]*seed.Snap, error) {
```

Comment:
hm this has become a bit hard to follow, could we try something like:
```
type systemAndEssentialSnaps struct {
    *System
    Seed seed.Seed,
    InfosByType map[snap.Type]*snapInfo
    SeedSnapsByType map[snap.Type]*seed.Snap
}
```

then this becomes:    
```suggestion
	*systemAndEssentialSnaps, error) {
```

Reply:
Good idea, I'll do this.

Reply:
Done!